packages/design-system/src/components/ChoiceList/Choice.jsx

```diff
@@ -25,13 +26,20 @@ export class Choice extends React.PureComponent {
       // Event emitters are only relevant for uncontrolled radio buttons
       if (this.props.type === 'radio') {
         this.uncheckEventName = `${this.props.name}-uncheck`;
-        dsChoiceEmitter.on(this.uncheckEventName, this.handleUncheck.bind(this));
+        dsChoiceEmitter.on(this.uncheckEventName, this.handleUncheck);
```

Comment:
Ahhhh! was it binding to an old reference of `this`?

Reply:
That took a long time to figure out that unbind from emitter `off` did not work because of `this.handleUncheck.bind(this)`. Once this got moved out, then the addition of unbind of emitter in componentWillUnmount finally works.

Reply:
Yes, the `checked` of null issue was due to the emitter containing the old dom references - we need to remove this reference each time the parent radio button is changed which triggers componentWillUnmount

Reply:
this looks like a good opportunity to use react hooks, what do you think?

Reply:
If Choice was a functional component, then yes, it would be good to use react hooks now. As it is presently a class component, i think it's better to just add the componentWillUnmount lifecycle to clean up the emitter event instead of rewriting as a functional component.

Reply:
Ah right, we cant rewrite this component because it uses refs anyway. I think in the future though in general if we are working on a component and we *can* easily modernize it, we should try to. Like maybe this doesn't have to be done in this PR, but I'm thinking on quick modernization could be to remove instances of `.bind` by adding [arrow functions](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties)